curp/src/channel/mod.rs

```diff
@@ -180,15 +180,13 @@ impl<KM> KeyBasedReceiverInner<KM> {
         let mut listener = None;
         loop {
             channel.flush_queue(&mut self.buf);
-            match self.buf.pop_front() {
```

Comment:
The clippy in CI has problem with these lines. But my local clippy keeps warning me `single_match_else` here. So I refactor these lines. I guess we need to update the clippy version in our CI

Reply:
You can try to check the clippy version locally.

Reply:
Can we fix the clippy issue?

Reply:
Fixed.